### PR TITLE
added queryMessageFlags method

### PR DESCRIPTION
**Pull request by gumuz**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

-

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -699,6 +728,11 @@ export type UserResponse<UserType = UnknownType> = User<UserType> & {
  * Option Types
  */
 
+export type MessageFlagsPaginationOptions = {
```
Comment (ferhatelmas) [inline src/types.ts:731]: looks like a base, don't we have it around ?

File: src/types.ts [hunk 2]
```diff
@@ -107,6 +107,35 @@ export type AppSettingsAPIResponse<
...
+    >;
+    user: UserResponse<UserType>;
+    approved_at?: string;
+    created_at?: string;
+    created_by_automod?: boolean;
+    moderation_result?: {};
+    rejected_at?: string;
+    reviewed_at?: string;
+    reviewed_by?: UserResponse<UserType>;
+    updated_at?: string;
```
Comment (ferhatelmas) [inline src/types.ts:135]: dates weren't optional on server

File: src/types.ts [hunk 3]
```diff
@@ -1049,6 +1083,12 @@ export type EventTypes =
 
 export type AscDesc = 1 | -1;
 
+export type MessageFlagsFilters = {
+  channel_cid?: string;
```
Comment (ferhatelmas) [inline src/types.ts:1087]: server supports `$in` too

File: src/client.ts [hunk 4]
```diff
@@ -1576,6 +1579,30 @@ export class StreamChat<
...
+   * @param {MessageFlagsFilters} filterConditions filter conditions
+   * @param {MessageFlagsPaginationOptions} options Option object, {limit: 10, offset:0}
+   *
+   * @return {Promise<MessageFlagsResponse<ChannelType, CommandType, UserType>>} Message Flags Response
+   */
+  async queryMessageFlags(
+    filterConditions: MessageFlagsFilters = {},
+    options: MessageFlagsPaginationOptions = {},
+  ) {
+    // Return a list of user bans
```
Comment (ferhatelmas) [inline src/client.ts:1594]: what is this comment ?

File: src/types.ts (hunk 1 above)
Comment (gumuz) [reply src/types.ts:732]: not that I'm aware of

**Review by ferhatelmas — APPROVED** [review]
LGTM

File: src/types.ts (hunk 2 above)
Comment (gumuz) [reply src/types.ts:135]: they are now

**Comment by ferhatelmas** [comment]
@mahboubii good to give a last look and merge.

**Comment by gumuz** [comment]
@mahboubii even though this is already merged, could you take a quick look at it to see if all is well and perhaps create a release for it? 🙏🏽

File: src/types.ts [hunk 5]
```diff
@@ -107,6 +107,35 @@ export type AppSettingsAPIResponse<
...
+      CommandType,
+      MessageType,
+      ReactionType,
+      UserType
+    >;
+    user: UserResponse<UserType>;
+    approved_at?: string;
+    created_at?: string;
+    created_by_automod?: boolean;
+    moderation_result?: {};
```
Comment (mahboubii) [inline src/types.ts:131]: this is not 100% accurate, could be `UnknownType` if we don't know the exact type

Comment (gumuz) [reply src/types.ts:131]: created a new PR for this here: https://github.com/GetStream/stream-chat-js/pull/681

**Pull request merged**